agent/security_helper.py

```diff
@@ -33,3 +33,36 @@ def is_firewall_enabled():
     filter_table = iptc.Table(iptc.Table.FILTER)
     input_chain = next(filter(lambda c: c.name == 'INPUT', filter_table))
     return not len(input_chain.rules) == 0
+
+
+def parse_line(l):
+    fields = l.split()
+    state = fields[5] if len(fields) > 5 and fields[5] != '-' else None
+    program = fields[6] if len(fields) > 6 and fields[6] != '-' else None
+    return {
+        'proto': fields[0],
+        'recv_q': fields[1],
+        'send_q': fields[2],
+        'local_address': fields[3],
+        'foreign_address': fields[4],
+        'state': state,
+        'program': program}
+
+
+def netstat_scan():
+    """
+    Returns all open inet connections with their addresses and PIDs.
+    """
+    out = netstat(['-pa'])
```

Comment:
Can we use some library or built-in for this? I don't like relying on `sh` for parsing. It's too fragile. I had to do it for `nmap` because I didn't find an alternative.

Reply:
@vpetersson We can use `psutil` [net_connections](https://psutil.readthedocs.io/en/latest/index.html?highlight=net_connections#psutil.net_connections)

Reply:
That looks like a much more robust solution.